platform-common/src/main/java/com/softicar/platform/common/io/stream/InputStreamCollection.java

```diff
@@ -0,0 +1,84 @@
+package com.softicar.platform.common.io.stream;
+
+import com.softicar.platform.common.core.exception.ExceptionsCollector;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.function.Supplier;
+
+/**
+ * Manages the safe opening and closing of a {@link Collection} of
+ * {@link InputStream} objects.
+ * <p>
+ * This class implements {@link AutoCloseable} and guarantees the proper closing
+ * of all successfully opened {@link InputStream}, as long as the {@link #close}
+ * method is called, which should be done by a try-block.
+ *
+ * @author Oliver Richers
+ */
+public class InputStreamCollection implements AutoCloseable {
+
+	private final Collection<InputStream> streams;
+
+	public InputStreamCollection() {
+
+		this.streams = new ArrayList<>();
+	}
+
+	/**
+	 * Uses the given {@link Supplier} to open the {@link InputStream} and adds
```

Comment:
> to open the {@link InputStream}

We can't be certain that this actually happens here. E.g. in the following scenario that "opening" would have already happened:

```
var streamCollection = new InputStreamCollection();
InputStream inputStream = getInputStream(); // somehow gets an InputStream from somewhere
streamCollection.add(() -> inputStream);
```

Since we cannot rule out that such a pattern is used, I think we need to reword the Javadoc.

Reply:
changed